src/condor_scripts/condor_upgrade_check

```diff
@@ -718,6 +718,43 @@ def check_desktop_policy():
     return 0
 
 
+def check_pslot_default():
+    """Check for use of RANK in startd config due to incompatibility with
+    P-Slots on by default in V23.0.0"""
+    htc_version = htcondor.param["CONDOR_VERSION"].split(".")
+    htc_version = tuple(int(x) for x in htcondor.param["CONDOR_VERSION"].split("."))
+    if htc_version >= (23, 0, 0):
+        return 0
+    for key in htcondor.param.keys():
+        if "SLOT_TYPE_" in key:
```

Comment:
This `key` object is now a (case-sensitive) `str`, so the comparison needs to be case normalized:
```suggestion
        if "slot_type_" in key.lower():
```
---
Using `str.startswith(...)` would also be more efficient, unless there are cases where `SLOT_TYPE_` shows up in the middle of the attr name:
```suggestion
        if key.lower().startswith("slot_type_"):
```